espressif/idf-eclipse-plugin
# IEP-369: Partition table editor

## Description

Added a partition table editor, which allows the creation and editing of partitions.csv files. The name of the file is getting from sdkconfig.

Fixes # ([IEP-369](https://jira.espressif.com:8443/browse/IEP-369))

## Type of change
- New feature (non-breaking change which adds functionality)

## How has this been tested?

Test 1:
- Create a new project,
 - right click on the project in project explorer -> ESP-IDF: Partition Table Editor
 - Information message should appear, because no sdkconfig file was found.
 
 Test 2:
 - build the project to generate sdkconfig file
 - open Partition Table Editor
 - you should see the default table, which you can edit
 - SubType Combobox values depend on the selected type. Values should corresponding to [this](https://docs.espressif.com/projects/esp-idf/en/latest/esp32/api-guides/partition-tables.html#subtype)
 
 Test 3:
 - open Partition Table Editor
 - Click "Add Row"
 - add some values that satisfy the validator.
 - Click "Save" -> the message "Saving successful" should appear.
 - close and reopen the dialog
 - values must match the last save
 
 Test 4:
 - open Partition Table Editor
 - Select some row
 - Click "Delete Selected" -> Confirmation message should appear -> click "Yes"
 - the row should be visually deleted from the table 
 - click "Save"
 - close and reopen the dialog
 - values must match the last save
 
 Test 5:
 - open Partition Table Editor
 - try to add some values that do not satisfy the validator. 
 - Cells with errors should be highlighted as on the screenshot: 
![image](https://user-images.githubusercontent.com/24419842/197196504-4eb81149-10f0-4042-bbd7-8900a102b87e.png)
- If you hover over a cell with an error, you should see a tooltip telling you what's wrong
- Click "Save" -> You should see an error message:
![image](https://user-images.githubusercontent.com/24419842/197196878-6f645568-3dc8-47d3-b785-9cb021b57907.png)
- fix errors and save again -> the message "Saving successful" should appear.



**Test Configuration**:
* ESP-IDF Version:
* OS (Windows,Linux and macOS):

## Checklist
